File: LeetCode/Dynamic Programming/198. House Robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:

        ln = len(nums)
        dp = {-1: 0, 0: nums[0]}
        for i in range(1, ln):

            pick = nums[i]
            if (i > 1):
                pick += dp[i-2]
            nonpick = dp[i-1]

            dp[i] = max(pick, nonpick)
           # print(pick,nonpick,dp)

        return dp[ln-1]

File: LeetCode/198.house-robber.cs
using System;
/*
 * @lc app=leetcode id=198 lang=csharp
 *
 * [198] House Robber
 */

// @lc code=start
public class Solution
{
    public int Rob(int[] nums)
    {
        if (nums.Length == 0)
        {
            return 0;
        }
        int prev1 = 0;
        int prev2 = 0;
        foreach (var i in nums)
        {
            int Current = Math.Max(prev1 + i, prev2);
            prev1 = prev2;
            prev2 = Current;
        }
        return prev2;

    }
}
// @lc code=end

